### PR TITLE
Add ESLint to enforce Google JS style in public/

### PR DESCRIPTION
**Pull request by prodonjs**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2579)

### CODE REVIEW

File: components/centraldashboard/public/components/main-page.js [hunk 1]
```diff
@@ -27,72 +27,82 @@ import '@polymer/neon-animation/animations/fade-out-animation.js';
...
-                ]
-            },
-            sidebarItemIndex: { type: Number, value: 0 },
-            primaryViewIndex: { type: Number, value: 0 },
-            homeOrIframeViewIndex: { type: Number, value: 0 },
-            url: { type: String, value: "" },
-            buildVersion: { type: String, value: VERSION},
-            _devMode: { type: Boolean, value: false },
-        }
+  /**
```
[the remark this answers is not in the document]
Comment (prodonjs) [reply components/centraldashboard/public/components/main-page.js]: Ack'ed

**Comment by prodonjs** [comment]
If you are using VSCode, it will be useful to add this to your workspace settings.

```
{
    "editor.formatOnSave": false,
    "eslint.enable": true,
    "eslint.autoFixOnSave": true,
    "eslint.workingDirectories": [
        "../public"
    ]
}
```

File: components/centraldashboard/webpack.config.js [hunk 2]
```diff
@@ -109,6 +123,12 @@ module.exports = {
         ])
     },
     optimization: {
+        minimizer: [new TerserPlugin({
```
Comment (avdaredevil) [inline components/centraldashboard/webpack.config.js]: LMGT's but what does this do?

Comment (prodonjs) [reply components/centraldashboard/webpack.config.js]: It handles minification. When used for a production build, it also strips out a `dist/public/vendor.bundle.js.LICENSE` file containing all of the license comments from the vendored dependencies.

**Pull request merged**